store/src/java/com/zimbra/cs/service/mail/ItemAction.java

```diff
@@ -100,12 +101,17 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         Element action = request.getElement(MailConstants.E_ACTION);
         String operation = action.getAttribute(MailConstants.A_OPERATION).toLowerCase();
 
-        String successes = handleCommon(context, request, operation, MailItem.Type.UNKNOWN);
+        ItemActionResult result = handleCommon(context, request, MailItem.Type.UNKNOWN);
 
         Element response = zsc.createElement(MailConstants.ITEM_ACTION_RESPONSE);
         Element act = response.addUniqueElement(MailConstants.E_ACTION);
-        act.addAttribute(MailConstants.A_ID, successes);
+        act.addAttribute(MailConstants.A_ID, Joiner.on(",").join(result.getSuccessIds()));
         act.addAttribute(MailConstants.A_OPERATION, operation);
+        String opStr = getOperation(operation);
+        if (opStr.equals(MailConstants.OP_HARD_DELETE))
+        {
+            act.addAttribute(MailConstants.A_NON_EXISTENT_IDS, Joiner.on(",").join(((DeleteActionResult)result).getNonExistentIds()));
```

Comment:
OK.  So only populating this for deletes.  I'm guessing we may end up extending this for other actions at some point in the future but probably sensible to keep the scope narrow for now.